Docs: Vue - Remove reference to Volar extension

Joe-Moran opened the pull request:
The link is broken and the Volar Vue (official) plugin is the only one needed now.

Related: https://github.com/vuejs/language-tools/issues/3939

## What I did

Removed the link and cleaned up grammar.

## Checklist for Contributors

### Testing

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

Change is to documentation only; no testing needed

### Documentation

<!-- Please check (put an "x" inside the "[ ]") the applicable items below to indicate which documentation has been updated. -->

- [x] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

<!-- CANARY_RELEASE_SECTION -->

<!-- BENCHMARK_SECTION -->
<!-- BENCHMARK_SECTION -->

<!-- greptile_comment -->

## Greptile Summary

This PR updates the TypeScript integration documentation to remove references to the deprecated TypeScript Vue Plugin VSCode extension. The change reflects current best practices in the Vue ecosystem, as Volar is now the only extension needed for Vue + TypeScript development in VSCode.

Specifically, the change:
- Removes the reference and link to the TypeScript Vue Plugin
- Updates the grammar in the documentation to maintain clarity
- Aligns with Vue's official tooling recommendations

## Confidence score: 5/5

1. This PR is very safe to merge as it only updates documentation to remove outdated information
2. The change is well-researched and backed by official Vue tooling recommendations (issue #3939)
3. Files that need attention:
   - docs/configure/integration/typescript.mdx - verify that no other Vue-related TypeScript documentation needs updating

<sub>1 file reviewed, no comments</sub>
<sub>[Edit PR Review Bot Settings](https://app.greptile.com/review/github) | [Greptile](https://greptile.com?utm_source=greptile_expert&utm_medium=github&utm_campaign=code_reviews&utm_content=storybook_32079)</sub>

<!-- /greptile_comment -->

Joe-Moran committed:
docs: remove deprecated vscode extension

jonniebigodes commented on docs/configure/integration/typescript.mdx:
```diff
@@ -96,7 +96,7 @@ Out of the box, Storybook supports the `satisfies` operator for almost every fra
   ### The TypeScript auto-completion is not working on my editor
 
-  If you're using Vue single file components and TypeScript, you can add both [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) and the [TypeScript Vue Plugin](https://marketplace.visualstudio.com/items?itemName=Vue.vscode-typescript-vue-plugin) for editor support, additional type safety and auto-completion. Nevertheless, if you're working with Svelte, you can add the [Svelte ...
+  If you're using Vue single file components and TypeScript, you can add [Volar](https://marketplace.visualstudio.com/items?itemName=Vue.volar) for editor support, additional type safety and auto-completion. Nevertheless, if you're working with Svelte, you can add the [Svelte for VSCode extension](https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode) for similar benefits.
```
@Joe-Moran when you can, could you adjust this to the following to ensure we're referencing the correct extension name:
```suggestion
  If you're using Vue single file components and TypeScript, you can add the official [Vue](https://marketplace.visualstudio.com/items?itemName=Vue.volar) extension for editor support, additional type safety and auto-completion. Nevertheless, if you're working with Svelte, you can add the [Svelte for VSCode extension](https://marketplace.visualstudio.com/items?itemName=svelte.svelte-vscode) for similar benefits.
```

jonniebigodes replied:
@Joe-Moran, one small item as well. I was wondering if you could adjust the [Writing Stories in TypeScript](https://github.com/storybookjs/storybook/blob/next/docs/writing-stories/typescript.mdx) and use the same naming to ensure consistency across the documentation. I saw the earlier pr that you opened.

jonniebigodes reviewed (changes requested):
@Joe-Moran thank you for taking the time to put together this pull request and helping us improve the documentation by catching this small issue. We appreciate it. 

I left a small item for you to look into when you can.

Let me know once you've addressed it.
Have a great day
Stay safe

Joe-Moran committed:
chore: apply suggested name change

Volar is not the main tool used anymore, it's now a Vue-specific extension built on Volar

Co-authored-by: jonniebigodes <joaocontadesenvolvimento@gmail.com>

Joe-Moran committed:
Merge branch 'storybookjs:next' into patch-4

Joe-Moran committed:
docs: update naming of extension from Volar to Vue

The official Vue extension is now built on top of Volar and doesn't use the Volar name

jonniebigodes committed:
Merge branch 'next' into patch-4

jonniebigodes reviewed (approved):
@Joe-Moran appreciate you addressing the feedback so promptly 🙏 ! 
All is good on my end and I'll gladly merge this once the checklist clears.

Hope you have a great day.

Stay safe
